Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1349,3 +1354,63 @@ private class BrowserScreenshotHelper: ScreenshotHelper {
         return nil
     }
 }
+
+private class BrowserPrintPageRenderer: UIPrintPageRenderer {
+    private weak var browser: Browser?
+    let textAttributes = [NSFontAttributeName: UIFont(name: "Helvetica", size: 8)!]
+    let dateString: String
+
+    init(browser: Browser) {
+        self.browser = browser
+        let dateFormatter = NSDateFormatter()
+        dateFormatter.dateStyle = .ShortStyle
+        dateFormatter.timeStyle = .ShortStyle
+        self.dateString = dateFormatter.stringFromDate(NSDate())
+
+        super.init()
+
+        self.footerHeight = 0.5 * 36
+        self.headerHeight = 0.5 * 36
+
+        let formatter = browser.webView.viewPrintFormatter()
+        formatter.perPageContentInsets = UIEdgeInsets(top: 36, left: 36, bottom: 36, right: 36)
+        addPrintFormatter(formatter, startingAtPageAtIndex: 0)
+    }
+
+    override func drawFooterForPageAtIndex(pageIndex: Int, var inRect headerRect: CGRect) {
+        var headerInsets = UIEdgeInsets(top: CGRectGetMinY(headerRect), left: 36, bottom: CGRectGetMaxY(paperRect) - CGRectGetMaxY(headerRect), right: 36)
+        headerRect = UIEdgeInsetsInsetRect(paperRect, headerInsets)
+
+        // url on left
+        self.drawAtPoint(browser!.displayURL?.absoluteString ?? "", rect: headerRect, onLeft: true)
```

Comment:
Part of me wants to make this a little more... configurable. i.e. we can have an array:
Header: [HeaderFields] = [HeaderField.Url, HeaderFile.None, HeaderField.PageNumber]

but until we have a way to configure it, that's useless! Maybe file a follow up to look into this sometime...